libsolidity/formal/Predicate.cpp

```diff
@@ -214,19 +215,64 @@ string Predicate::formatSummaryCall(
 	}
 
 	/// The signature of a function summary predicate is: summary(error, this, abiFunctions, cryptoFunctions, txData, preBlockChainState, preStateVars, preInputVars, postBlockchainState, postStateVars, postInputVars, outputVars).
-	/// Here we are interested in preInputVars to format the function call,
-	/// and in txData to retrieve `msg.value`.
+	/// Here we are interested in preInputVars to format the function call.
 
-	string value;
-	if (auto v = readTxVars(_args.at(4)).at("msg.value"))
+	string txModel;
+
+	if (_appendTxVars)
 	{
-		bigint x(*v);
-		if (x > 0)
-			value = "{ value: " + *v + " }";
+		set<string> txVars;
+		if (isFunctionSummary() && programFunction()->isPayable())
+			txVars.insert("msg.value");
```

Comment:
So the assumption is that `programFunction()` is not NULL when it's a normal function but it could be NULL if it's a constructor? I think there should be an assert against this:
```suggestion
		if (isFunctionSummary())
		{
			smtAssert(programFunction(), "");
			if (programFunction()->isPayable())
				txVars.insert("msg.value");
		}
```

Reply:
If it's an actual constructor it should be the same as a normal function, but if it's a contract and that contract has a constructor this should also be true.

Reply:
Added assertion.